packages/relayer-lib/Cargo.toml

```diff
@@ -16,21 +16,32 @@ ibc-eureka-solidity-types = { workspace = true, features = ["rpc"] }
 sp1-ics07-tendermint-prover   = { workspace = true, optional = true }
 sp1-ics07-tendermint-utils    = { workspace = true, optional = true }
 
-serde = { workspace = true, features = ["derive"] }
-prost = { workspace = true, features = ["std"] }
+ethereum-apis = { workspace = true }
+ethereum-light-client = { workspace = true }
+ethereum-types = { workspace = true }
 
-async-trait = { workspace = true }
-anyhow      = { workspace = true, features = ["std"] }
-futures     = { workspace = true, default-features = true }
-tracing     = { workspace = true, default-features = true }
+serde      = { workspace = true, features = ["derive"] }
+prost      = { workspace = true, features = ["std"] }
+serde_json = { workspace = true }
+
+async-trait   = { workspace = true }
+anyhow        = { workspace = true, features = ["std"] }
+futures       = { workspace = true, default-features = true }
+futures-timer = { workspace = true }
+tracing       = { workspace = true, default-features = true }
 
 tendermint     = { workspace = true, features = ["std"] }
 tendermint-rpc = { workspace = true, features = ["http-client"] }
 tendermint-light-client-verifier = { workspace = true }
 
 ibc-proto-eureka    = { workspace = true }
 ibc-core-host-types = { workspace = true }
+ibc-core-commitment-types = { workspace = true }
 
-alloy          = { workspace = true, features = ["full", "node-bindings"] }
+alloy-primitives    = { workspace = true }
+alloy-provider      = { workspace = true }
+alloy-sol-types     = { workspace = true }
+alloy-transport     = { workspace = true }
+alloy-rpc-types-eth = { workspace = true }
```

Comment:
I think it would be better to use `alloy` here. This is why `"full"` and `"node-bindings"` features exist. To make our deps simpler when possible

Reply:
Sure, I'm OK with that. I don't have a very strong opinion on that. It's more of a personal preference to to only bring in exactly the dependencies I need, rather than get a lot of extra stuff and features we might not need. But if you think using `alloy` is better, we can certainly do that.

Reply:
Ref: https://github.com/cosmos/solidity-ibc-eureka/pull/190/files#r1902195076